storage/local.go

```diff
@@ -70,29 +62,27 @@ func (local *LocalBackend) Get(ctx context.Context, url string, hostPath string,
 
 // PutFile copies a file from the hostPath into storage.
 func (local *LocalBackend) PutFile(ctx context.Context, url string, hostPath string) error {
-	path, ok := getPath(url)
+	path := getPath(url)
+	return linkFile(hostPath, path)
+}
 
+// Supports indicates whether this backend supports the given storage request.
+// For the LocalBackend, the url must start with "file://" be in an allowed directory
+func (local *LocalBackend) Supports(rawurl string) error {
+	path := getPath(rawurl)
+	ok := strings.HasPrefix(path, "/")
 	if !ok {
-		return fmt.Errorf("local storage does not support put on %s", url)
+		return fmt.Errorf("Must provided an absolute path")
 	}
-
 	if !isAllowed(path, local.allowedDirs) {
-		return fmt.Errorf("Can't access file, path is not in allowed directories:  %s", url)
+		return fmt.Errorf("Can't access file, path is not in allowed directories:  %s", path)
```

Comment:
This should stay `url` (or `rawurl`) so that the error relates directly to the task message, without transformation.